### PR TITLE
Merge MCCI changes into repo

### LINKED ISSUE #2: Merge with the current version from local
Push the local changes to repo and sync to version 1.0.1 which is the current version

### PR DESCRIPTION
**Pull request by sjpark-mcci**
fix #2: sync with v1.0.1 from local

### CODE REVIEW

File: boards.txt [hunk 1]
```diff
@@ -15,222 +15,77 @@
...
+mcci_catena_4410.build.mcu=cortex-m0plus
+mcci_catena_4410.build.f_cpu=48000000L
+mcci_catena_4410.build.usb_product="Catena 4410"
+mcci_catena_4410.build.usb_manufacturer="MCCI"
+mcci_catena_4410.build.board=SAMD_FEATHER_M0
+mcci_catena_4410.build.core=arduino
+mcci_catena_4410.build.extra_flags=-DARDUINO_SAMD_ZERO -D__SAMD21G18A__ {build.usb_flags}
+mcci_catena_4410.build.ldscript=linker_scripts/gcc/flash_with_bootloader.ld
+mcci_catena_4410.build.openocdscript=openocd_scripts/arduino_zero.cfg
+mcci_catena_4410.build.variant=arduino_zero
```
Comment (terrillmoore) [inline boards.txt]: This should be either Catena4410 or feather, not arduino_zero, I think. And yes, we should have a variant directory for each board.

File: boards.txt [hunk 2]
```diff
@@ -15,222 +15,77 @@
...
+mcci_catena_4410.upload.protocol=sam-ba
+mcci_catena_4410.upload.maximum_size=262144
+mcci_catena_4410.upload.use_1200bps_touch=true
+mcci_catena_4410.upload.wait_for_upload_port=true
+mcci_catena_4410.upload.native_usb=true
+mcci_catena_4410.build.mcu=cortex-m0plus
+mcci_catena_4410.build.f_cpu=48000000L
+mcci_catena_4410.build.usb_product="Catena 4410"
+mcci_catena_4410.build.usb_manufacturer="MCCI"
+mcci_catena_4410.build.board=SAMD_FEATHER_M0
```
Comment (terrillmoore) [inline boards.txt]: This really should be SAMD_MCCI_CATENA_4410

File: boards.txt [hunk 3]
```diff
@@ -15,222 +15,77 @@
...
+mcci_catena_4450.build.mcu=cortex-m0plus
+mcci_catena_4450.build.f_cpu=48000000L
+mcci_catena_4450.build.usb_product="Catena4450"
+mcci_catena_4450.build.usb_manufacturer="MCCI"
+mcci_catena_4450.build.board=SAMD_FEATHER_M0_EXPRESS
+mcci_catena_4450.build.core=arduino
+mcci_catena_4450.build.extra_flags=-DARDUINO_SAMD_ZERO -DARDUINO_SAMD_FEATHER_M0 -DMCCI_CATENA_4450 -D__SAMD21G18A__ {build.usb_flags}
+mcci_catena_4450.build.ldscript=linker_scripts/gcc/flash_with_bootloader.ld
+mcci_catena_4450.build.openocdscript=openocd_scripts/arduino_zero.cfg
+mcci_catena_4450.build.variant=feather_m0_express
```
Comment (terrillmoore) [inline boards.txt]: Seems wrong

File: boards.txt [hunk 4]
```diff
@@ -15,222 +15,77 @@
...
+mcci_catena_4450.build.variant_system_lib=
+mcci_catena_4450.build.vid=0x040E
+mcci_catena_4450.build.pid=0x0019
```
Comment (terrillmoore) [inline boards.txt]: seems wrong, see comment above about `build.pid`.

File: boards.txt [hunk 5]
```diff
@@ -15,222 +15,77 @@
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
 
-adafruit_feather_m0.name=Adafruit Feather M0
-adafruit_feather_m0.vid.0=0x239A
```
Comment (terrillmoore) [inline boards.txt]: Why do they have vid.0/pid.0 and vid.1/pid.1. Don't we need that, too?

File: boards.txt [hunk 6]
```diff
@@ -15,222 +15,77 @@
...
-adafruit_itsybitsy_m0.build.extra_flags=-DCRYSTALLESS -DADAFRUIT_ITSYBITSY_M0 -D__SAMD21G18A__ {build.usb_flags}
-adafruit_itsybitsy_m0.build.ldscript=linker_scripts/gcc/flash_with_bootloader.ld
-adafruit_itsybitsy_m0.build.openocdscript=openocd_scripts/arduino_zero.cfg
-adafruit_itsybitsy_m0.build.variant=itsybitsy_m0
-adafruit_itsybitsy_m0.build.variant_system_lib=
-adafruit_itsybitsy_m0.build.vid=0x239A
-adafruit_itsybitsy_m0.build.pid=0x800F
-adafruit_itsybitsy_m0.bootloader.tool=openocd
-adafruit_itsybitsy_m0.bootloader.file=itsybitsy/samd21_sam_ba.bin
+mcci_catena_4460.name=Catena4460
```
Comment (terrillmoore) [inline boards.txt]: This should be `MCCI Catena 4460`, not mashed together like this. See Adafruit examples

File: boards.txt [hunk 7]
```diff
@@ -15,222 +15,77 @@
...
+mcci_catena_4460.pid.0=0x001A
+mcci_catena_4460.upload.tool=bossac
+mcci_catena_4460.upload.protocol=sam-ba
+mcci_catena_4460.upload.maximum_size=262144
+mcci_catena_4460.upload.use_1200bps_touch=true
+mcci_catena_4460.upload.wait_for_upload_port=true
+mcci_catena_4460.upload.native_usb=true
+mcci_catena_4460.build.mcu=cortex-m0plus
+mcci_catena_4460.build.f_cpu=48000000L
+mcci_catena_4460.build.usb_product="Catena4460"
```
Comment (terrillmoore) [inline boards.txt]: This must be `"MCCI Catena 4460"` -- it's a UI string.

File: boards.txt [hunk 8]
```diff
@@ -15,222 +15,77 @@
...
+mcci_catena_4460.build.usb_manufacturer="MCCI"
```
Comment (terrillmoore) [inline boards.txt]: This should be "MCCI Corporation" -- it's a UI string.

File: boards.txt [hunk 9]
```diff
@@ -15,222 +15,77 @@
...
+mcci_catena_4460.build.board=SAMD_FEATHER_M0_EXPRESS
```
Comment (terrillmoore) [inline boards.txt]: Really seems wrong. We shoudn't be using SAMD_FEATHER_M0_EXPRESS, as this is not an EXPRESS.

File: cores/arduino/SERCOM.cpp [hunk 10]
```diff
@@ -652,20 +652,16 @@ void SERCOM::initClockNVIC( void )
     clockId = GCM_SERCOM3_CORE;
     IdNvic = SERCOM3_IRQn;
   }
-#if defined(SERCOM4)
```
Comment (terrillmoore) [inline cores/arduino/SERCOM.cpp]: We shouldn't do it this way. Instead we need to look at defining SERCOM4 and 5 -- they have to be defined, or this code wouldn't compile.

File: cores/arduino/WInterrupts.c [hunk 11]
```diff
@@ -62,48 +62,23 @@ void attachInterrupt(uint32_t pin, voidFuncPtr callback, uint32_t mode)
 #else
   EExt_Interrupts in = digitalPinToInterrupt(pin);
 #endif
-  if (in == NOT_AN_INTERRUPT) return;
+  if (in == NOT_AN_INTERRUPT || in == EXTERNAL_INT_NMI)
```
Comment (terrillmoore) [inline cores/arduino/WInterrupts.c]: Why did we make this change? Or did we possibly miss a bug fix that is in my repo?

File: cores/arduino/WInterrupts.c [hunk 12]
```diff
@@ -62,48 +62,23 @@ void attachInterrupt(uint32_t pin, voidFuncPtr callback, uint32_t mode)
...
+    return;
 
   if (!enabled) {
     __initialize();
     enabled = 1;
   }
 
-  if (in == EXTERNAL_INT_NMI) {
```
Comment (terrillmoore) [inline cores/arduino/WInterrupts.c]: I think we missed an important bug fix/improvement.

File: cores/arduino/WInterrupts.c [hunk 13]
```diff
@@ -62,48 +62,23 @@ void attachInterrupt(uint32_t pin, voidFuncPtr callback, uint32_t mode)
...
-
-  } else { // Not NMI, is external interrupt
-
     // Enable wakeup capability on pin in case being used during sleep
     EIC->WAKEUP.reg |= (1 << in);
 
     // Assign pin to EIC
     pinPeripheral(pin, PIO_EXTINT);
 
+  // Assign callback to interrupt
```
Comment (terrillmoore) [inline cores/arduino/WInterrupts.c]: Why did we move this? Or did we miss a bug fix?

File: boards.txt [hunk 14]
```diff
@@ -15,222 +15,77 @@
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
 
-adafruit_feather_m0.name=Adafruit Feather M0
-adafruit_feather_m0.vid.0=0x239A
-adafruit_feather_m0.pid.0=0x800B
```
Comment (terrillmoore) [inline boards.txt]: I'm not sure we should be deleting the feather_m0.... this code is useful to more people than MCCI. Instead we should just add our extra material.

File: cores/arduino/wiring_analog tempsensor [hunk 15]
```diff
@@ -1,460 +0,0 @@
-/*
```
Comment (terrillmoore) [inline cores/arduino/wiring_analog tempsensor]: looks like we missed an enhancement.

File: post_install.bat [hunk 16]
```diff
@@ -1,10 +0,0 @@
-@echo off
```
Comment (terrillmoore) [inline post_install.bat]: Don't we need this on Win7?

File: variants/feather_m0/debug_scripts/variant.gdb [hunk 17]
```diff
@@ -1,31 +0,0 @@
-#
```
Comment (terrillmoore) [inline variants/feather_m0/debug_scripts/variant.gdb]: We *definitely* want to keep this!  This is what we need for debugging with gdb and openocd. Please revert, and add similar for each of our boards.

File: variants/feather_m0/linker_scripts/gcc/flash_with_bootloader.ld [hunk 18]
```diff
@@ -1,216 +0,0 @@
-/*
```
Comment (terrillmoore) [inline variants/feather_m0/linker_scripts/gcc/flash_with_bootloader.ld]: We need these for each of MCCI's boards, and we don't want to remove the Adafruit ones.

**Comment by sjpark-mcci** [comment]
These changes are from v1.0.1 package. I thought you wanted to sync this repository to the current version first and work from there. Are you suggesting that we should not merge yet but fix more things and make it as v1.0.2?

**Comment by terrillmoore** [comment]
It’s a bit painful to delete and then revert. Perhaps we should not remove the files that are being removed. Anyway, I’ll think more about it, it’s not a blocking item so we can wait another day.

 

--Terry

**Comment by terrillmoore** [comment]
I guess it's time to move forward on this, because it's blocking progress on the Catena 4470. I'll go over the review one more time, but @sjpark-mcci please don't let this get dropped...

**Comment by sjpark-mcci** [comment]
Thank you, I will test on my side as well.

**Pull request merged**